agent/backend/pktvisor/pktvisor.go

```diff
@@ -273,96 +273,108 @@ func (p *pktvisorBackend) Start() error {
-				SchemaVersion: fleet.CurrentRPCSchemaVersion,
-				Func:          fleet.AgentMetricsRPCFunc,
-				Payload:       batchPayload,
+		var batchPayload []fleet.AgentMetricsRPCPayload
+		totalSize := 0
+		for pName, pMetrics := range metrics {
+			data, err := p.policyRepo.GetByName(pName)
+			if err != nil {
+				p.logger.Error("skipping pktvisor policy not managed by orb", zap.String("policy", pName), zap.Error(err))
+				continue
 			}
-
-			body, err := json.Marshal(rpc)
+			payloadData, err := json.Marshal(pMetrics)
 			if err != nil {
-				p.logger.Error("error marshalling metric rpc payload", zap.Error(err))
-				return
+				p.logger.Error("error marshalling scraped metric json", zap.String("policy", pName), zap.Error(err))
+				continue
 			}
-
-			if token := p.mqttClient.Publish(p.metricsTopic, 1, false, body); token.Wait() && token.Error() != nil {
-				p.logger.Error("error sending metrics RPC", zap.String("topic", p.metricsTopic), zap.Error(token.Error()))
+			metricPayload := fleet.AgentMetricsRPCPayload{
+				PolicyID:   data.ID,
+				PolicyName: data.Name,
+				Datasets:   data.GetDatasetIDs(),
+				Format:     "json",
+				BEVersion:  p.pktvisorVersion,
+				Data:       payloadData,
 			}
-			p.logger.Info("scraped and published metrics", zap.String("topic", p.metricsTopic), zap.Int("payload_size_b", totalSize), zap.Int("batch_count", len(batchPayload)))
+			batchPayload = append(batchPayload, metricPayload)
+			totalSize += len(payloadData)
+			p.logger.Info("scraped metrics for policy", zap.String("policy", pName), zap.String("policy_id", data.ID), zap.Int("payload_size_b", len(payloadData)))
+		}
 
-		})
+		rpc := fleet.AgentMetricsRPC{
+			SchemaVersion: fleet.CurrentRPCSchemaVersion,
+			Func:          fleet.AgentMetricsRPCFunc,
+			Payload:       batchPayload,
+		}
 
+		body, err := json.Marshal(rpc)
 		if err != nil {
-			return err
+			p.logger.Error("error marshalling metric rpc payload", zap.Error(err))
+			return
+		}
+
+		if token := p.mqttClient.Publish(p.metricsTopic, 1, false, body); token.Wait() && token.Error() != nil {
+			p.logger.Error("error sending metrics RPC", zap.String("topic", p.metricsTopic), zap.Error(token.Error()))
 		}
+		p.logger.Info("scraped and published metrics", zap.String("topic", p.metricsTopic), zap.Int("payload_size_b", totalSize), zap.Int("batch_count", len(batchPayload)))
 
-		job.SingletonMode()
+	})
+
+	if err != nil {
+		return err
 	}
 
+	job.SingletonMode()
 	return nil
 }
 
+func (p *pktvisorBackend) scrapeOpentelemetry(err error) {
+	ctx := context.Background()
+	p.exporter, err = createExporter(ctx, p.logger)
+	if err != nil {
+		p.logger.Error("failed to create a exporter", zap.Error(err))
+	}
+	p.receiver, err = createReceiver(ctx, p.exporter, p.logger)
+	if err != nil {
+		p.logger.Error("failed to create a receiver", zap.Error(err))
+	}
+
+	err = p.exporter.Start(ctx, nil)
+	if err != nil {
+		p.logger.Error("otel exporter startup error", zap.Error(err))
+		os.Exit(1)
```

Comment:
let's get rid of these `os.Exit()` calls and these `p.logger.Error` messages and instead do a `return err` the same as `scrapeDefault`, and let the agent startup handle the error